src/main/java/com/danielagapov/spawn/Services/ChatMessage/ChatMessageService.java

```diff
@@ -55,22 +65,67 @@ public List<ChatMessageDTO> getChatMessagesByTagId(UUID tagId) {
 
     public ChatMessageDTO saveChatMessage(ChatMessageDTO chatMessageDTO) {
         try {
-            // Fetch User and Event entities
             User userSender = userRepository.findById(chatMessageDTO.userSenderId())
                     .orElseThrow(() -> new BaseNotFoundException(chatMessageDTO.userSenderId()));
             Event event = eventRepository.findById(chatMessageDTO.eventId())
                     .orElseThrow(() -> new BaseNotFoundException(chatMessageDTO.eventId()));
 
-            // Convert DTO to Entity
             ChatMessage chatMessageEntity = ChatMessageMapper.toEntity(chatMessageDTO, userSender, event);
 
-            // Save Entity
             chatMessageRepository.save(chatMessageEntity);
 
-            // Convert Entity back to DTO and return
             return ChatMessageMapper.toDTO(chatMessageEntity);
         } catch (DataAccessException e) {
             throw new BaseSaveException("Failed to save chatMessage: " + e.getMessage());
         }
     }
+
+    public void createChatMessageLike(UUID chatMessageId, UUID userId) {
+        try {
+            boolean exists = chatMessageLikesRepository.existsByChatMessage_IdAndUser_Id(chatMessageId, userId);
+            if (exists) {
+                throw new BaseSaveException("Like already exists for chatMessageId: " + chatMessageId + " and userId: "
+                        + userId);
+            }
+            ChatMessage chatMessage = chatMessageRepository.findById(chatMessageId)
+                    .orElseThrow(() -> new BaseSaveException("ChatMessageId: " + chatMessageId));
+            User user = userRepository.findById(userId)
+                    .orElseThrow(() -> new BaseSaveException("UserId: " + userId));
+
+            ChatMessageLikes chatMessageLikes = new ChatMessageLikes();
+            chatMessageLikes.setChatMessage(chatMessage);
+            chatMessageLikes.setUser(user);
+
+            chatMessageLikesRepository.save(chatMessageLikes);
+
+        } catch (Exception e) {
+            throw new BaseSaveException("Like: chatMessageId: " + chatMessageId + " userId: "
+                    + userId + ". Error: " + e.getMessage());
+        }
+    }
+
+    public List<UserDTO> getChatMessageLikes(UUID chatMessageId) {
+
+        ChatMessage chatMessage = chatMessageRepository.findById(chatMessageId)
+                .orElseThrow(() -> new BaseNotFoundException(chatMessageId));
+
+        List<ChatMessageLikes> likes = chatMessageLikesRepository.findByChatMessage(chatMessage);
+
+        return likes.stream().map(like -> UserMapper.toDTO(like.getUser())).collect(Collectors.toList());
+    }
+
+    public void deleteChatMessageLike(UUID chatMessageId, UUID userId) {
+        try {
+            boolean exists = chatMessageLikesRepository.existsByChatMessage_IdAndUser_Id(chatMessageId, userId);
+            if (!exists) {
+                throw new BaseDeleteException("Like not found for chatMessageId: " + chatMessageId + " and userId: "
```

Comment:
Split these two types of errors into two proper types of errors, instead of them both being `BaseDeleteException`s with different messages. Maybe: `LikeNotFoundException` and `LikeDeletionException`

Reply:
I abstracted `BaseNotFoundException` rather than making new exceptions since I noticed a lot of other sections were also using it.